substrate/frame/support/procedural/src/pallet/expand/call.rs

```diff
@@ -255,23 +255,34 @@ pub fn expand_call(def: &mut Def) -> proc_macro2::TokenStream {
 	let default_deprecation_info =
 		quote::quote! { #frame_support::__private::metadata_ir::DeprecationStatus::NotDeprecated }
 			.to_string();
-
-	let indexes: Vec<proc_macro2::TokenStream> = methods
-		.iter()
-		.filter_map(|x| {
-			let key = x.call_index;
-			let deprecation_status =
-				crate::deprecation::get_deprecation(&quote::quote! {#frame_support}, &x.attrs)
+	let deprecation = {
```

Comment:
This looks good here, a few suggestions to make things a bit easier to follow:
- turn `get_deprecation` panic into a compile_error!
- is there a way to remove the `default_deprecation_info`? Maybe `unwarp_or_else`? 
- we could double check if the `deprecation` attribute appears both on the `enum` as `FullyDeprecated`, as well as on the individual variants:

```rust
#[deprecated(..)]
enum Call {
    #[deprecated(..)]
    VariantA(..)
}
```
We could return a compile error to ask the developer to pick one or the other, since the overall `deprecated` on the enum negates the need of individual deprecation

Reply:
Nice, I'll implement this